.clang-tidy

```diff
@@ -0,0 +1,12 @@
+Checks:
+  *,
+  -altera*,
+  -llvm*,
+  -fuchsia-default-arguments-*,
+  -bugprone-easily-swappable-parameters,
+  -cppcoreguidelines-avoid-magic-numbers,
+  -modernize-use-trailing-return-type,
+  -readability-magic-numbers,
+CheckOptions:
+  - { key: hicpp-signed-bitwise.IgnorePositiveIntegerLiterals, value: 1 }
```

Comment:
Hmmm... does this permit bitwise operations on signed literals? Are positive int literals by default unsigned? I think we should complain unless the user marks it `+1u` or something.

### Reference
[1] https://releases.llvm.org/10.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/hicpp-signed-bitwise.html

Reply:
I'm actually not too sure whether this check is useful. This SO post has some good discussion: https://stackoverflow.com/questions/50399090/use-of-a-signed-integer-operand-with-a-binary-bitwise-operator-when-using-un.

I think the int promotion thing raised in the post is a good point, so I'm inclined to remove that `IgnorePositiveIntegerLiterals` option.